lib/rest-client.js

```diff
@@ -19,13 +19,17 @@ module.exports = CoreObject.extend({
     });
   },
 
-  upload: function(keyPrefix, revisionKey, value) {
+  upload: function(keyPrefix, revisionKey, revisionData, value) {
     var uploadKey = keyPrefix + ':' + revisionKey;
     var requestBody = {
       id: uploadKey,
       body: value
     };
 
+    if(revisionData) {
+      requestBody.revision_data = JSON.stringify(revisionData);
```

Comment:
Why is `JSON.stringify` needed here? And correspondingly, the JSON.parse below.

My guess is that your API is expecting a string and not an embedded object. But to me that seems like it should be fixed in the API, not worked around in the client. What's your thinking?